File: java/structure/flyWeight/src/ConcreteFlyWeightSimpleFactory.java
public class ConcreteFlyWeightSimpleFactory {
    private static ConcreteFlyWeightSimpleFactory instance;

    public static  ConcreteFlyWeightSimpleFactory GetFactory() {
        if (instance == null) {
            synchronized (ConcreteFlyWeightSimpleFactory.class) {
                if (instance == null) {
                    instance = new ConcreteFlyWeightSimpleFactory();
                }
            }
        }
        return instance;
    }


    public FlyWeight GetConcreteFlyWeight(String key) {
        switch (key) {
            case "ConcreteFlyWeightFirst":
                return new ConcreteFlyWeightFirst();
            case "ConcreteFlyWeightSecond":
                return new ConcreteFlyWeightSecond();
            default:
                return null;
        }
    }

}

File: go/structuretype/flyWeight/ConcreteFlyWeightSimpleFactory.go
package flyWeight

import "sync"

var (
	concreteFlyWeightSimpleFactory *ConcreteFlyWeightSimpleFactory
	onceConcreteFlyWeightFactory   sync.Once
)

type ConcreteFlyWeightSimpleFactory struct{}

// GetConcreteFlyWeightSimpleFactory 单例模式创建唯一实例
func GetConcreteFlyWeightSimpleFactory() *ConcreteFlyWeightSimpleFactory {
	onceConcreteFlyWeightFactory.Do(func() {
		concreteFlyWeightSimpleFactory = &ConcreteFlyWeightSimpleFactory{}
	})
	return concreteFlyWeightSimpleFactory
}

func (concreteFlyWeightSimpleFactory *ConcreteFlyWeightSimpleFactory) Get(key string) ConcreteFlyWeight {
	switch key {
	case "ConcreteFlyWeightFirst":
		return &ConcreteFlyWeightFirst{}
	case "ConcreteFlyWeightSecond":
		return &ConcreteFlyWeightSecond{}
	default:
		return nil
	}
}
